pkg/frontend/querymiddleware/codec_test.go

```diff
@@ -99,14 +133,310 @@ func TestMetricsQueryRequest(t *testing.T) {
 
 			req, err := codec.DecodeMetricsQueryRequest(ctx, r)
 			if err != nil || tc.expectedErr != nil {
-				require.EqualValues(t, tc.expectedErr, err)
+				require.Equal(t, tc.expectedErr, err)
 				return
 			}
-			require.EqualValues(t, tc.expected, req)
+			requireEqualMetricsQueryRequest(t, tc.expected, req)
 
 			rdash, err := codec.EncodeMetricsQueryRequest(context.Background(), req)
 			require.NoError(t, err)
-			require.EqualValues(t, tc.url, rdash.RequestURI)
+			require.Equal(t, tc.url, rdash.RequestURI)
+		})
+	}
+}
+
+func TestMetricsQuery_MinMaxTime(t *testing.T) {
+
+	startTime, err := time.Parse(time.RFC3339, "2024-02-21T00:00:00-08:00")
+	require.NoError(t, err)
+	endTime, err := time.Parse(time.RFC3339, "2024-02-22T00:00:00-08:00")
+	require.NoError(t, err)
+
+	atModifierDuration := 10 * time.Minute
+
+	stepDurationStr := "60s"
+	stepDuration, _ := time.ParseDuration(stepDurationStr)
+
+	rangeVectorDurationStr := "5m"
+	rangeVectorDuration, _ := time.ParseDuration(rangeVectorDurationStr)
+	rangeVectorDurationMS := rangeVectorDuration.Milliseconds()
+
+	offsetDurationStr := "1h"
+	offsetDuration, _ := time.ParseDuration(offsetDurationStr)
+	offsetDurationMS := offsetDuration.Milliseconds()
+
+	rangeRequest := NewPrometheusRangeQueryRequest(
+		"/api/v1/query_range",
+		startTime.UnixMilli(),
+		endTime.UnixMilli(),
+		stepDuration.Milliseconds(),
+		time.Duration(0),
```

Comment:
We should set the lookback period to something non-zero to ensure that it is correctly considered when determining the min / max times.

(same applies for instant query in this test, and both instant and range queries in `TestMetricsQuery_MinMaxTime_TransformConsistency`)